slru/slru.go

```diff
@@ -0,0 +1,140 @@
+package slru
+
+import (
+	"container/list"
+)
+
+type segment int
+
+const (
+	probation segment = iota
+	protected
+)
+
+// Cache is a segmented LRU cache. It is not safe for concurrent access.
+type Cache struct {
+	items        map[string]*list.Element
+	probation    *list.List
+	protected    *list.List
+	maxProbation int
+	maxProtected int
+}
+
+type entry struct {
+	key     []byte
+	value   []byte
+	segment segment
```

Comment:
This could just as easily be a pointer to its containing list, which would be just as expressive, though I worry about giving the GC more work to do in breaking cycles.